pkg/jobrunaggregator/jobrunaggregatoranalyzer/pass_fail_test.go

```diff
@@ -0,0 +1,146 @@
+	jobRunIDToAvailabilityResultForBackend := make(map[string]jobrunaggregatorlib.AvailabilityResult)
+	for i, disruption := range disruptions {
+		jobRunIDToAvailabilityResultForBackend[fmt.Sprintf("run_%d", i)] = jobrunaggregatorlib.AvailabilityResult{SecondsUnavailable: disruption}
+	}
+
+	return jobRunIDToAvailabilityResultForBackend
+}
+
+func TestCheckPercentileDisruption(t *testing.T) {
+	weeklyAverageFromTenDays := weeklyAverageFromTenDays{}
+
+	tests := []struct {
+		name                 string
+		disruptions          []int
+		thresholdPercentile  int
+		historicalDisruption float64
+		status               testCaseStatus
+		failedCount          int
+		successCount         int
+	}{
+		{
+			// Required Passes for 95th percentile is 6
+			// no disruption so it passes
+			name:                 "Test 95th Percentile 0's Pass",
+			disruptions:          []int{0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
+			thresholdPercentile:  95,
+			historicalDisruption: 7,
+			status:               testCasePassed,
+			failedCount:          0,
+			successCount:         10,
+		},
+		{
+			// Required Passes for 95th percentile is 6
+			// 5 Natural Passes
+			// Multiplier (6 - 5) == 1
+			// Fuzz Threshold ((historicalDisruption / 5) + 1) = 2
+			// The Disruption value that == 7 gets flipped to pass
+			name:                 "Test 95th Percentile Fuzzy Pass",
+			disruptions:          []int{5, 5, 5, 6, 6, 7, 9, 9, 9, 9},
+			thresholdPercentile:  95,
+			historicalDisruption: 6,
+			status:               testCasePassed,
+			failedCount:          4,
+			successCount:         6,
+		},
+		{
+			// Required Passes for 95th percentile is 6
+			// 4 Natural Passes
+			// Multiplier (6 - 4) == 2
+			// Fuzz Threshold ((historicalDisruption / 5) + 1) = 2
+			// The Disruption values that == 7 get flipped to passes
+			name:                 "Test 95th Percentile Multi Fuzzy Pass",
+			disruptions:          []int{5, 5, 5, 6, 7, 7, 8, 8, 8, 86},
+			thresholdPercentile:  95,
+			historicalDisruption: 6,
+			status:               testCasePassed,
+			failedCount:          4,
+			successCount:         6,
+		},
+		{
+			// Required Passes for 95th percentile is 6
+			// 4 Natural Passes
+			// Multiplier (6 - 4) == 2
+			// Fuzz Threshold ((historicalDisruption / 5) + 1) = 2
+			// The Disruption value that == 7 gets flipped to pass
+			// But the 8s and above do not, so we don't have enough passes
+			name:                 "Test 95th Percentile Multi Fuzzy Fail",
+			disruptions:          []int{5, 5, 5, 6, 7, 8, 8, 9, 9, 86},
+			thresholdPercentile:  95,
+			historicalDisruption: 6,
+			status:               testCaseFailed,
+			failedCount:          5,
+			successCount:         5,
+		},
+		{
+			// Required Passes for 80th percentile is 4
+			// 4 Natural Passes
+			name:                 "Test 80th Percentile Pass",
+			disruptions:          []int{0, 0, 1, 1, 2, 2, 2, 2, 2, 2},
```

Comment:
Were these tests based on real world data?

Reply:
There were based on cases where I saw us fail by 1 missing required job off by 1 second.  But not real data from one of those runs.  I can add that.